docs/articles_en/learn-openvino/llm_inference_guide/genai-guide/genai-use-cases.rst

```diff
@@ -0,0 +1,296 @@
+GenAI Use Cases
+=====================
+
+This article provides several use case scenarios for Generative AI model
+inference. The applications presented in the code samples below
+only require minimal configuration, like setting an inference device. Feel free
+to explore and modify the source code as you need.
+
+
+Using GenAI for Text-to-Image Generation
+#######################################
+
+Examples below demonstrate inference on text-to-image models, like Stable Diffusion
+1.5, 2.1, and LCM, with a text prompt as input. The :ref:`main.cpp <maincpp>`
+sample shows basic usage of the ``Text2ImagePipeline`` pipeline.
+:ref:`lora.cpp <loracpp>` shows how to apply LoRA adapters to the pipeline.
+
+
+.. tab-set::
+
+   .. tab-item:: C++
+      :sync: cpp
+
+      .. tab-set::
+
+         .. tab-item:: main.cpp
+            :name: maincpp
+
+            .. code-block:: cpp
+
+               int32_t main(int32_t argc, char* argv[]) try {
+                   OPENVINO_ASSERT(argc == 3, "Usage: ", argv[0], " <MODEL_DIR> '<PROMPT>'");
+
+                   const std::string models_path = argv[1], prompt = argv[2];
+                   const std::string device = "CPU";  // GPU, NPU can be used as well.
+
+                   ov::genai::Text2ImagePipeline pipe(models_path, device);
+                   ov::Tensor image = pipe.generate(prompt,
+                       ov::genai::width(512),
+                       ov::genai::height(512),
+                       ov::genai::num_inference_steps(20),
+                       ov::genai::num_images_per_prompt(1));
+
+                   // Saves images with a `num_images_per_prompt` name pattern.
+                   imwrite("image_%d.bmp", image, true);
+
+                   return EXIT_SUCCESS;
+               } catch (const std::exception& error) {
+                   try {
+                       std::cerr << error.what() << '\n';
+                   } catch (const std::ios_base::failure&) {}
+                   return EXIT_FAILURE;
+               } catch (...) {
```

Comment:
I guess this is a nice opportunity to remove try catch which is normally required by SDL.

Reply:
I have updated the code snippets:

https://github.com/openvinotoolkit/openvino/pull/27062/commits/08e4a066c128c5e0ce4daca9d6e14488132070ba